#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_embed_initial_visibility.dpatch by  <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix gtkmozembed's EmbedWindow::GetVisibility. bz#312998

@DPATCH@

=== embedding/browser/gtk/src/EmbedWindow.cpp
==================================================================
--- xulrunner/embedding/browser/gtk/src/EmbedWindow.cpp	(revision 77)
+++ xulrunner/embedding/browser/gtk/src/EmbedWindow.cpp	(local)
@@ -359,7 +359,10 @@
 NS_IMETHODIMP
 EmbedWindow::GetVisibility(PRBool *aVisibility)
 {
-  *aVisibility = mVisibility;
+  *aVisibility = mVisibility ||
+                 !mOwner->mIsChrome &&
+                 mOwner->mOwningWidget &&
+                 GTK_WIDGET_MAPPED(mOwner->mOwningWidget);
   return NS_OK;
 }
 
